src/common/NonEmptyList.ml

```diff
@@ -0,0 +1,20 @@
+open Core_kernel
+
+type 'a t = NonEmpty of 'a * 'a list
```

Comment:
To quote Alan Perlis, "It is better to have 100 functions operate on one data structure than to have 10 functions operate on 10 data structures."

Can we stick with list? I don't think the non-empty guarantee is worth writing another list library.

Reply:
Non-empty list has always been used in validation, I just pulled it out to a separate module. `NonEmptyList` is also a private module of common so isn't exposed. That said, the API to `Validation` guarantees you can't get to an error state without at least one error and we always convert to list so it isn't adding a lot.

Reply:
Removed `NonEmptyList`